R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,256 @@
+#' Derive a Clinical Benefit Parameter
+#'
+#' Add a clinical benefit/disease control parameter to the input dataset.
+#'
+#' @details
+#' Clinical benefit/disease control is first identified for looking for subjects
+#' having objective response, and then derived for subjects that have at least
+#' one evaluable non-PD response assessment after a specified amount of time
+#' from a reference date.
+#'
+#' @param dataset Input dataset. This is the dataset to which the clinical
+#' benefit rate parameter will be added.
+#'
+#'   The variables `PARAMCD`, `AVALC`, `ADT`, and those specified by the `by_vars`
+#'   parameter are expected.
+#'
+#' @param dataset_adsl ADSL dataset used as input for populating `subject_keys`
+#' in input dataset.
```

Comment:
This sounds strange because the `subject_keys` should be in the input dataset. The dataset is the reference for the observations to create. I.e., for each subject in `dataset_adsl` an observation is created for the new parameter even if the subject has no observations in the input dataset.

Reply:
updated